dd-java-agent/agent-profiling/profiling-controller-openjdk/src/main/java/com/datadog/profiling/controller/openjdk/events/SmapEntryFactory.java

```diff
@@ -91,34 +115,54 @@ private static HashMap<Long, String> getAnnotatedRegions() {
         String[] emptyStringArgs = {};
         Object[] dcmdArgs = {emptyStringArgs};
         String[] signature = {String[].class.getName()};
-
         String[] lines =
             ((String) mbs.invoke(objectName, "systemMap", dcmdArgs, signature)).split("\n");
         HashMap<Long, String> annotatedRegions = new HashMap<>();
+        boolean arrivedAtMappings = false;
 
         for (String line : lines) {
-          Matcher matcher = SYSTEM_MAP_ENTRY_PATTERN.matcher(line);
-          if (matcher.matches()) {
-            long startAddress;
-            if (matcher.group(1).equals("0x" + VSYSCALL_START_ADDRESS)) {
-              // See how smap entry parsing is done for vsyscall
-              startAddress = -0x1000 - 1;
-            } else {
-              startAddress = Long.decode(matcher.group(1));
+          // The output begins with a legend followed by a `size prot offset What` table header
+          if (!arrivedAtMappings) {
+            if (line.startsWith("size")) {
+              arrivedAtMappings = true;
             }
-            String description = matcher.group(6);
-            annotatedRegions.put(startAddress, description);
-            if (description.isEmpty()) {
-              annotatedRegions.put(startAddress, "UNDEFINED");
-            } else if (description.startsWith("STACK")) {
+            continue;
+          } else {
+            // The final line contains the total number of mappings and their net vsize
+            if (line.startsWith("Total")) {
+              break;
+            }
+          }
+
+          String[] segments = line.split("\\s+");
```

Comment:
This will expand to full regex - I think this should do the same thing
```suggestion
          String segment = "";
          int idx = line.indexOf(' ');
          if (idx > - 1) {
            segment = line.substring(0, idx);
          }
          ...
```